### PR TITLE
feat: Add persistent volume claim, affinity and tolerations support for clickhouse to the helm chart.

### PR DESCRIPTION
**Pull request by jredl-va**
By default clickhouse is provisioned with standard gcp persistent disk. We swapped out to SSD disks shortly after our usage of clickhouse ramped up.  

This pr will allow users to define their own volume claim to attach to clickhouse. In addition this pull request exposes the ability to set affinity / tolerations and resources for the clickhouse instance provisioned by the clickhouse operator. 

These instructions probably belong somewhere else but I've included them here for reference. 

gcloud example for provisioning a disk:

```
gcloud compute disks create pvc-clickhouse --type=pd-ssd --size=2048GB --zone=us-central1-c
```

To create a persistent volume claim that targets this disk:
```
# This creates a volume claim using the same name specified within the clickhouse values file
apiVersion: v1
kind: PersistentVolume
metadata:
  name: clickhouse-volume
spec:
  persistentVolumeReclaimPolicy: Retain
  storageClassName: ""
  capacity:
    storage: 2048Gi
  accessModes:
    - ReadWriteOnce
  gcePersistentDisk:
    pdName: pvc-clickhouse
    fsType: ext4
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: clickhouse-pvc
spec:
  # It's necessary to specify "" as the storageClassName
  # so that the default storage class won't be used, see
  # https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
  storageClassName: ""
  volumeName: clickhouse-volume
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 2048Gi
```

values to target a node pool for posthog (larger resources as clickhouse wants more memory / cpu) and set the volume claim
```
clickhouse:
  # -- Optional: Used to manually specify a persistent volume claim. When specified the cloud specific storage class will not be provisioned
  persistentVolumeClaim: "clickhouse-pvc"
  # -- Affinity settings for clickhouse pod
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
          - matchExpressions:
              - key: cloud.google.com/gke-nodepool
                operator: In
                values:
                  - posthog-analytics
  # -- Toleration labels for clickhouse pod assignment
  tolerations:
    - effect: NoSchedule
      key: posthog-analytics
      operator: Equal
      value: "true"
  # -- Clickhouse resource requests/limits. See more at http://kubernetes.io/docs/user-guide/compute-resources/
  resources:
    requests:
      memory: "50Gi"
    limits:
      memory: "50Gi"
```

For your consideration:
@macobo @fuziontech @tiina303

### CODE REVIEW

**Comment by jredl-va** [comment]
Hrm...will need to close and reopen. Somehow I lost a commit on my side.

**Comment by jredl-va** [comment]
Found my missed commits....reopening.

File: charts/posthog/templates/clickhouse_instance.yaml [hunk 1]
```diff
@@ -100,8 +118,14 @@ spec:
                 - name: interserver
                   containerPort: 9009
               volumeMounts:
+              {{- if (eq (include "clickhouse.persistentVolumeClaim" . ) "" ) }}
                 - name: data-volumeclaim-template
```
Comment (tiina303) [inline charts/posthog/templates/clickhouse_instance.yaml:122]: I wonder if we can always use `clickhouse-storage` name and simplify by dropping this condition here (& renaming on line 145), but I don't know what the name change would do for upgrades 😅 so ignore this.

**Review by tiina303 — APPROVED** [review]
I'm fine with landing this as is.

> These instructions probably belong somewhere else but I've included them here for reference.

Yes, in the [gcp page](https://posthog.com/docs/self-host/deploy/gcp) could just be section in the end. If you're willing to put up a pr against our docs too that would be awesome, the file is [here](https://github.com/PostHog/posthog.com/blob/master/contents/docs/self-host/deploy/gcp.mdx), but I can do that as well.

File: charts/posthog/templates/_helpers.tpl [hunk 2]
```diff
@@ -242,6 +242,16 @@ Set clickhouse fullname
 {{- end -}}
 {{- end -}}
```
[the remark this answers is not in the document]
Comment (jredl-va) [reply charts/posthog/templates/_helpers.tpl:244]: @tiina303 thanks! I like that much more. Just applied those changes.

**Review by fuziontech — APPROVED** [review]
Thank you so much for doing this. This is perfect for managing and tuning the underlying disk for clickhouse!

**Pull request merged**